packages/expo-cli/src/credentials/provider/AndroidCredentialsProvider.ts

```diff
@@ -78,19 +80,30 @@ export default class AndroidCredentialsProvider implements CredentialsProvider {
   public async getCredentialsAsync(
     src: CredentialsSource.LOCAL | CredentialsSource.REMOTE
   ): Promise<AndroidCredentials> {
+    let credentials: AndroidCredentials;
+
     switch (src) {
       case CredentialsSource.LOCAL:
-        return await this.getLocalAsync();
+        credentials = await this.getLocalAsync();
+        break;
       case CredentialsSource.REMOTE:
-        return await this.getRemoteAsync();
+        credentials = await this.getRemoteAsync();
+        break;
+    }
+
+    if (!this.options.skipCredentialsCheck) {
+      await validateKeystoreAsync(credentials.keystore);
     }
+
+    return credentials;
   }
 
   private async getRemoteAsync(): Promise<AndroidCredentials> {
     await runCredentialsManager(
       this.ctx,
       new SetupAndroidKeystore(this.projectFullName, {
         allowMissingKeystore: false,
+        skipKeystoreValidation: false,
```

Comment:
you are passing `skipKeystoreValidation` here but few lines vellow you have `isValidKeystore` call. Try to unify that validation.

btw this behaviour is correct we want perform that validation `isValidKeystore` but this implementation is confusing